python/pyiceberg/catalog/__init__.py

```diff
@@ -133,23 +133,6 @@ def __init__(self, name: str, **properties: str):
         self.name = name
         self.properties = properties
 
-    def property(self, key: str) -> str:
```

Comment:
I think it's best to remove this. There are places that bypass this check because the property is optional and `properties` are exposed to subclasses anyway. Since the subclasses need to pay attention to how `properties` are used, I think they should do that directly.